sweep/txgenerator.go

```diff
@@ -0,0 +1,328 @@
+package sweep
+
+import (
+	"fmt"
+	"github.com/btcsuite/btcwallet/wallet/txrules"
+
+	"sort"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+// TxGenerator generates sets of sweep txes.
+type TxGenerator struct {
+	// Signer is used by the sweeper to generate valid witnesses at the
+	// time the incubated outputs need to be spent.
+	Signer lnwallet.Signer
+
+	// MaxInputsPerTx specifies the  maximum number of inputs allowed in a
+	// single sweep tx. If more need to be swept, multiple txes are created
+	// and published.
+	MaxInputsPerTx int
+
+	dustLimit int64
+}
+
+func newTxGenerator(relayFeePerKW lnwallet.SatPerKWeight) *TxGenerator {
+	dustLimit := txrules.GetDustThreshold(
+		lnwallet.P2WPKHSize,
+		btcutil.Amount(relayFeePerKW.FeePerKVByte()))
+
+	return &TxGenerator{
+		dustLimit: int64(dustLimit),
+		// TODO: Initialize fields
+	}
+}
+
+// Generate goes through all given inputs and constructs sweep transactions,
```

Comment:
Mismatch with the comment and the method: it returns a set of inputs for one or many transactions. Perhaps we can introduce a new type to make this more explicit? 
```go
type CandidateTransaction []Input
```

The method would then instead return `[]CandidateTransaction`.

Reply:
Created type `inputSet` for this.